wfe/wfe.go

```diff
@@ -740,8 +742,17 @@ func (wfe *WebFrontEndImpl) RevokeCertificate(ctx context.Context, logEvent *req
 		return
 	}
 
-	// Use revocation code 0, meaning "unspecified"
-	err = wfe.RA.RevokeCertificateWithReg(ctx, *parsedCertificate, 0, registration.ID)
+	reason := core.RevocationCode(0)
+	if revokeRequest.Reason != nil && wfe.AcceptRevocationReason {
+		if _, present := core.UserRevocationReasons[*revokeRequest.Reason]; !present {
+			logEvent.AddError("invalid revocation reason code provided")
+			wfe.sendError(response, logEvent, probs.Malformed("invalid revocation reason code provided"), nil)
```

Comment:
Can you add a test case for the invalid revocation reason codepath? I don't think it gets exercised in `wfe_test.go` anywhere.